app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -46,15 +46,21 @@ def create
                               else
                                 "#{number_invitations_sent} Invitations Sent."
                               end
-    redirect_to "/chat_channel_memberships/#{@chat_channel.chat_channel_memberships.where(user_id: current_user).first&.id}/edit"
+    redirect_to edit_chat_channel_membership_path(@chat_channel.chat_channel_memberships.where(user_id: current_user).first&.id)
   end
 
-  def remove_invitation
+  def remove_membership
     @chat_channel = ChatChannel.find(params[:chat_channel_id])
     authorize @chat_channel, :update?
-    flash[:settings_notice] = "Invitation Removed."
-    ChatChannelMembership.where(chat_channel_id: @chat_channel.id, id: params[:invitation_id], status: "pending").first&.destroy
-    redirect_to "/chat_channel_memberships/#{@chat_channel.chat_channel_memberships.where(user_id: current_user).first&.id}/edit"
+    @chat_channel_membership = ChatChannelMembership.find(params[:membership_id])
```

Comment:
I am pretty sure we need to keep the `where(chat_channel_id: @chat_channel.id)` in this lookup to prevent anything nefarious from occurring. As is, someone could send a valid chat channel they are allowed to update but then manually change the  membership_id which could allow them to remove any membership they want.

Reply:
updated it to
`@chat_channel_membership = @chat_channel.chat_channel_memberships.find(params[:membership_id])`